### ISSUE TITLE
[R] CRAN packaging checklist for 14.0.0

### ISSUE THREAD

**Issue by paleolimbot** [apache/arrow#38141]
### Describe the enhancement requested

cc @assignUser @thisisnic 

## Before the release candidate is cut

- [x] [Create a GitHub issue](https://github.com/apache/arrow/issues/new/) entitled `[R] CRAN packaging checklist for version X.X.X` and copy this checklist to the issue.
- [x] Evaluate the status of any failing [nightly tests and nightly packaging builds](http://crossbow.voltrondata.com). These checks replicate most of the checks that CRAN runs, so we need them all to be passing or to understand that the failures may (though won't necessarily) result in a rejection from CRAN.
- [x] Check [current CRAN check results](https://cran.rstudio.org/web/checks/check_results_arrow.html)
- [x] Ensure the contents of the README are accurate and up to date.
- [x] Run `urlchecker::url_check()` on the R directory at the release candidate.
  commit. Ignore any errors with badges as they will be removed in the CRAN release branch.
- [x] [Polish NEWS](https://style.tidyverse.org/news.html#news-release) but do **not** update version numbers (this is done automatically later). You can find commits by, for example, `git log --oneline aa057d0..HEAD | grep "\[R\]"`
- [x] Run preliminary reverse dependency checks using `archery docker run r-revdepcheck`.
- [ ] For major releases, prepare tweet thread highlighting new features.

Wait for the release candidate to be cut:

## After release candidate has been cut

- [x] Create a CRAN-release branch from the release candidate commit

## Prepare and check the .tar.gz that will be released to CRAN.

- [x] `git fetch upstream && git checkout release-X.X.X-rcXX && git clean -f -d`
- [x] run `Rscript tools/update-checksums.R X.X.X` to download shasums for the precompiled binaries
- [x] Run `make build`. This copies Arrow C++ into tools/cpp, prunes some
  unnecessary components, and runs `R CMD build` to generate the source tarball.
  Because this will install the package, you will need to ensure that the version
  of Arrow C++ available to the configure script is the same as the version
  that is vendored into the R package (e.g., you may need to unset `ARROW_HOME`).
- [x] `devtools::check_built("arrow_X.X.X.tar.gz")` locally
- [x] Run reverse dependency checks using `archery docker run r-revdepcheck`.

## Release vote

- [x] Release vote passed!

## Generate R package to submit to CRAN

- [x] If the release candidate commit updated, rebase the CRAN release branch
  on that commit.
- [x] Pick any commits that were made to main since the release commit that
  were needed to fix CRAN-related submission issues identified in the above
  steps.
- [x] Remove badges from README.md
- [x] Run `urlchecker::url_check()` on the R directory
- [x] [Create a PR entitled `WIP: [R] Verify CRAN release-10.0.1-rc0`.](https://github.com/apache/arrow/pull/38682) Add
  a comment `@github-actions crossbow submit --group r` to run all R crossbow
  jobs against the CRAN-specific release branch.
- [x] Regenerate arrow_X.X.X.tar.gz (i.e., `make build`)

Ensure linux and macos binary packages are available:

- [x] Ensure linux binaries are available in the artifactory:
  https://apache.jfrog.io/ui/repos/tree/General/arrow/r

## Check binary Arrow C++ distributions specific to the R package

- [x] Upload the .tar.gz to [win-builder](https://win-builder.r-project.org/upload.aspx) (r-devel only)
  and confirm (with Nic, who will automatically receive an email about the results) that the check is clean.
- [x] Upload the .tar.gz to [MacBuilder](https://mac.r-project.org/macbuilder/submit.html)
  and confirm that the check is clean
- [x] Check `install.packages("arrow_X.X.X.tar.gz")` on Ubuntu and ensure that the
  hosted binaries are used
- [x] `devtools::check_built("arrow_X.X.X.tar.gz")` locally one more time (for luck)

## CRAN submission

- [x] Upload arrow_X.X.X.tar.gz to the
  [CRAN submit page](https://xmpalantir.wu.ac.at/cransubmit/)
- [x] Confirm the submission email

Wait for CRAN...
- [x] Accepted!
- [x] Tag the tip of the CRAN-specific release branch
- [ ] Add a new line to the matrix in the [backwards compatability job](https://github.com/apache/arrow/blob/main/dev/tasks/r/github.linux.arrow.version.back.compat.yml)
- [ ] (patch releases only) Update the package version in `ci/scripts/PKGBUILD`, `dev/tasks/homebrew-formulae/autobrew/apache-arrow.rb`, `r/DESCRIPTION`, and `r/NEWS.md`
- [ ] (CRAN-only releases) Rebuild news page with `pkgdown::build_news()` and submit a PR to the asf-site branch of the docs site with the contents of `arrow/r/docs/news/index.html` replacing the current contents of `arrow-site/docs/r/news/index.html`
- [ ] (CRAN-only releases) Bump the version number in `r/pkgdown/assets/versions.json`, and update this on the [the `asf-site` branch of the docs site](https://github.com/apache/arrow-site) too.
- [ ] Update the packaging checklist template to reflect any new realities of the
  packaging process.
- [ ] Wait for CRAN-hosted binaries on the
  [CRAN package page](https://cran.r-project.org/package=arrow) to reflect the
  new version
- [ ] Tweet!

### Component(s)

R

**Comment by thisisnic** [comment]
Thanks for kicking this off @paleolimbot!  I'm happy to prep the social media stuff a bit later on.

**Comment by assignUser** [comment]
We can remove all of the autobrew related stuff right? As well as the rtools-packages PRs as we are never looking for a local version of the win library unless the envvar is set...

**Comment by thisisnic** [comment]
Yeah, we should create a new ticket to update these instructions later, but I figured we might want to get 14.0.0 released first so we're aware of any other changes we need to make, and update it all in one PR.

**Comment by paleolimbot** [comment]
I think all the autobrew stuff is already gone (I just opened this issue before that merged)

**Comment by paleolimbot** [comment]
We've fixed our nightlies and our reverse dependency checks are clear! We've picked up 9 new reverse dependencies since 13.0.0 (for a total of 49!).

**Comment by assignUser** [comment]
I added the checksum script (though this is only necessary now (see #38205 ). The advanced refactor remving winlibs is https://github.com/apache/arrow/commit/2a4e0db7dd23d0aef6136e361472cba2bff8af21#diff-ee05cb2be40729ac4ed9b086fa4b8493347bd079b3c8944fc84877c4ed413181

**Comment by thisisnic** [comment]
Macbuilder on release branch with URLs fixed: https://mac.r-project.org/macbuilder/results/1699847336-f93888c5bca3f09c/

**Comment by thisisnic** [comment]
Macbuilder with fixes for Winbuilder: https://mac.r-project.org/macbuilder/results/1699899373-13af7a74c51dd423/
Winbuilder with fixes for Winbuilder:

**Comment by thisisnic** [comment]
`devtools::check_built()` resulted in a warning for:

```
❯ checking compilation flags used ... WARNING
  Compilation used the following non-portable flag(s):
    ‘-Wno-noexcept-type’ ‘-msse4.2’
  including flag(s) suppressing warnings
```

I could have sworn we only used to get 1 of these. Is this a blocker to submitting?

**Comment by assignUser** [comment]
That doesn't happen on the macbuilder so it should be ok (you probably have `arrow_r_dev` set?)

**Comment by thisisnic** [comment]
OK, great, submitted now!

**Comment by thisisnic** [comment]
Cherry-picked #38716 across

**Comment by assignUser** [comment]
https://github.com/apache/arrow/issues/38752 is the fix for the 'Additional issues' error

**Comment by assignUser** [comment]
https://github.com/apache/arrow/commit/c6682f11fa7b66e146fa999f79eae457f23f6628 adds more timeout so the windows failure we had doesn't happen again when the artifactory is a little slow.

**Comment by thisisnic** [comment]
Once #38819 is merged, we should cherr-pick that across too

**Comment by assignUser** [comment]
Merged: https://github.com/apache/arrow/commit/490cd83482594792de7e4d32ce83e8d4d56e1b96

@thisisnic that should be all for now, we will have to see how it goes...

**Comment by assignUser** [comment]
https://github.com/apache/arrow/commit/eb5de184a7e5d02f98526332ace54250417bd232 fixes an issue when (cross-)compiling arrow with strict linker options

**Comment by thisisnic** [comment]
@assignUser I've already submitted 14.0.0.1 now. We could include it in 14.0.0.2 but I suppose it's a question of whether we should.

**Comment by assignUser** [comment]
A RC for 14.0.2 will be cut this week so I am adding all of the cherrypicked issues (including the cmake change as there is no issue adding that to a new release :tada: )
